autobahn/src/main/java/io/crossbar/autobahn/wamp/messages/Subscribe.java

```diff
@@ -57,8 +58,21 @@ public static Subscribe parse(List<Object> wmsg) {
 
         long request = MessageUtil.parseRequestID(wmsg.get(1));
         Map<String, Object> options = (Map<String, Object>) wmsg.get(2);
-        String match = (String) options.get("match");
-        boolean getRetained = (boolean) options.get("get_retained");
+        
+        String match = null;
+        if (options.containsKey("match")) {
+            match = (String) options.get("match");
+            if (!Objects.equals(match, MATCH_EXACT) && !Objects.equals(match, MATCH_PREFIX) &&
+                    !Objects.equals(match, MATCH_WILDCARD)) {
+                throw new ProtocolError("match must be one of exact, prefix or wildcard.");
+            }
+        }
+        
+        boolean getRetained = false;
+        if (options.containsKey("get_retained")) {
```

Comment:
Using options.getOrDefault() would be a cleaner approach here.

Reply:
You are right. Didn't know about this method. I will change it.